controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -524,8 +525,11 @@ public void getScalingEvents(final String scopeName, final String streamName, fi
             }
 
             if (referenceEvent != null) {
-                finalScaleMetadataList.add(0, referenceEvent);
+                finalScaleMetadataList.add(referenceEvent);
             }
+
+            // Sort all the entries according to timestamp such that reference event is right in front.
+            finalScaleMetadataList.sort(Comparator.comparingLong(ScaleMetadata::getTimestamp));
```

Comment:
Is this sort going to be expensive for long-running streams?

Reply:
It is a function of range specified in this api (`from` and `to`). So if the range is small then number of records will also be small. Otherwise if its a large record, we should  be looking at pagination and hence limit number of records. 

Another point is, this list would typically be sorted already.